README.md

```diff
@@ -1,84 +1,143 @@
-## blocklock-solidity
+# blocklock-solidity
 
-This repository contains the Solidity-based smart contracts library that facilitates Randamu's on-chain timelock encryption and decryption.
+[![Solidity ^0.8.x](https://img.shields.io/badge/Solidity-%5E0.8.x-blue)](https://soliditylang.org/)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green)](LICENSE)
+[![Foundry Tests](https://img.shields.io/badge/Tested%20with-Foundry-red)](https://book.getfoundry.sh/)
 
-By leveraging this library, developers can implement time-based data unlocking mechanisms securely in their smart contracts.
+A Solidity library enabling on-chain blocklock encryption and decryption, from the [dcipher threshold network](https://dcipher.network/). This facilitates secure time-based data unlocking mechanisms for smart contracts.
 
-This library is designed with modularity and simplicity in mind, allowing developers to extend and integrate it into their existing projects easily.
+## ✨ Overview
+
+Controlling access to data based on time is crucial for various use cases, such as auctions, voting, and content release schedules. `blocklock-solidity` provides developers with tools to implement blocklock encryption on-chain, ensuring that encrypted data can only be decrypted after a specified block height, thus enhancing security and fairness in time-sensitive operations. 
+
+This blocklock library is powered by the dciper threshold network using **BLS pairing-based signature scheme** and **identity-based encryption** to achieve data encryption toward a future block height without relying on a trusted third party.  It is especially useful in decentralized settings where there is no such trusted third party to enforce timing.
+
+The library is designed with modularity and simplicity in mind, allowing developers to easily integrate it into their existing smart contract projects to achieve blocklock on-chain. Its extensible architecture makes it suitable for a wide range of applications that require robust on-chain randomness.
```

Comment:
I dunno if 'achieving blocklock' is an established term (yet!). Are we assuming that devs know what blocklock means when they hit this repo?

Reply:
>robust on-chain randomness.

this doesn't provide randomness, that's https://github.com/randa-mu/randomness-solidity